src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1217,6 +1219,50 @@ public override int GetHashCode()
             return (this.Name.GetHashCode());
         }
 
+        internal static unsafe int GetIgnoreCaseHash(string source)
+        {
+            // Do not allocate on the stack if string is empty
+            if (source.Length == 0)
+            {
+                return source.GetHashCode();
```

Comment:
>GetHashCode [](start = 30, length = 11)

does this return different value with different runs? if not, then we can just return the value instead of this call.

Reply:
> does this return different value with different runs? 

Yes, it should.

Reply:
should we cache it then? I am not sure if it is worth recalculating the same value every time

---
In reply to: [174640851](https://github.com/dotnet/coreclr/pull/16945#discussion_r174640851) [](ancestors = 174640851)

Reply:
It is worth caching it (unclear that it is worth it), it should be cached by `String.GetHashCode`. Optimization independent on this PR.